src/Components/WebAssembly/Authentication.Msal/src/Interop/AuthenticationService.ts

```diff
@@ -0,0 +1,363 @@
+        }
+
+        try {
+            // Evaluate the promise to capture any authentication errors
+            await new Promise<Msal.AuthResponse>(promiseFactory);
+            // See https://github.com/AzureAD/microsoft-authentication-library-for-js/wiki/FAQs#q6-how-to-avoid-page-reloads-when-acquiring-and-renewing-tokens-silently
+            if (window !== window.parent && !window.opener) {
+                return this.operationCompleted();
+            } else {
+                const state = await this.retrieveState(callbackUrl);
+                return this.success(state);
+            }
+        } catch (e) {
+            if (this.isMsalError(e)) {
+                return this.error(e.errorMessage);
+            } else {
+                return this.error(e);
+            }
+        }
+    }
+
+    private isMsalError(resultOrError: any): resultOrError is Msal.AuthError {
+        return resultOrError?.errorCode;
+    }
+
+    private error(message: string) {
+        return { status: AuthenticationResultStatus.Failure, errorMessage: message };
+    }
+
+    private success(state: any) {
+        return { status: AuthenticationResultStatus.Success, state };
+    }
+
+    private redirect() {
+        return { status: AuthenticationResultStatus.Redirect };
+    }
+
+    private operationCompleted() {
+        return { status: AuthenticationResultStatus.OperationCompleted };
+    }
+}
+
+export class AuthenticationService {
+
+    static _infrastructureKey = 'Microsoft.Authentication.WebAssembly.Msal';
+    static _initialized = false;
+    static instance: MsalAuthorizeService;
+
+    public static async init(settings: AuthorizeServiceConfiguration) {
+        if (!AuthenticationService._initialized) {
+            AuthenticationService._initialized = true;
+            AuthenticationService.instance = new MsalAuthorizeService(settings);
+        }
+    }
+
+    public static getUser() {
+        return AuthenticationService.instance.getUser();
+    }
+
+    public static getAccessToken(request: AccessTokenRequestOptions) {
+        return AuthenticationService.instance.getAccessToken(request);
+    }
+
+    public static signIn(state: any) {
+        return AuthenticationService.instance.signIn(state);
+    }
+
+    // url is not used in the msal.js implementation but we keep it here
+    // as it is part of the default RemoteAuthenticationService contract implementation.
+    // The unused parameter here just reflects that.
+    public static completeSignIn(url: string) {
+        return AuthenticationService.instance.completeSignIn();
+    }
+
+    public static signOut(state: any) {
+        return AuthenticationService.instance.signOut(state);
+    }
+
+    public static completeSignOut(url: string) {
+        return AuthenticationService.instance.completeSignOut(url);
+    }
+}
+
+declare global {
+    interface Window { AuthenticationService: AuthenticationService; }
+}
+
+window.AuthenticationService = AuthenticationService;
```

Comment:
Is this the standard global we use for all the Blazor authentication services, or is this something specific to MSAL?

If we're adding a global for Blazor specifically, we might want to consider moving this inside the `Blazor` namespace (e..g, `Blazor.AuthenticationService`) so we're only impacting global scope in one way. Not saying this PR should be blocked for that, but want to check if my understanding is right!

Reply:
Ah!!

that's a great catch!!

I'm going to put each one of them in their own 'namespace' so that they don't conflict. I'll do a small change in the oidc-library so that its somewhere like window['Microsoft.AspNetCore.Components.WebAssembly.Authentication.AuthenticationService'] and similar for the msal one.

Reply:
Nevermind, it has to be the same unless we want to do something more advanced with the remote authentication service.

I'll file a follow-up issue, as I don't want to try and solve at this point.